staging/src/k8s.io/component-base/version/version.go

```diff
@@ -134,15 +136,15 @@ func (m *effectiveVersion) Validate() []error {
 	emulationVersion := m.emulationVersion.Load()
 	minCompatibilityVersion := m.minCompatibilityVersion.Load()
 
-	// emulationVersion can only be 1.{binaryMinor-1}...1.{binaryMinor}.
+	// emulationVersion can only be 1.{binaryMinor-3}...1.{binaryMinor}
 	maxEmuVer := binaryVersion
-	minEmuVer := binaryVersion.SubtractMinor(1)
+	minEmuVer := binaryVersion.SubtractMinor(3)
```

Comment:
Can we open an issue to move this logic out of component-base as well?  I would expect k8s components (kube-apiserver, ...) to set the limit according to the rules of that component (N-3 for k8s components), but leave it up to 3rd party component owners to decide what the limit should be for their components.

Reply:
That's a good idea. Would `staging/src/k8s.io/apimachinery/pkg/version` be a reasonable place to put it?

Reply:
Yes, and this PR doesn't need to address that problem now since it's preexisting, it just highlights that we should track and fix in a follow up

Reply:
Opened https://github.com/kubernetes/kubernetes/issues/129291 as a follow up